cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```diff
@@ -19,17 +19,27 @@
 import co.cask.cdap.api.dataset.table.Scanner;
 import co.cask.cdap.data2.dataset2.lib.table.FuzzyRowFilter;
 import co.cask.cdap.data2.dataset2.lib.table.MetricsTable;
-import com.google.common.collect.Lists;
 
 import java.io.IOException;
-import java.util.List;
 import java.util.Map;
 import java.util.NavigableMap;
+import java.util.Set;
 import java.util.SortedMap;
 import javax.annotation.Nullable;
 
 /**
- * Combined Metrics table to read from both v2 and v3 Tables
+ * Combined Metrics table to read from both v2 and v3 Tables and write to v3 table
+ *
+ * Writing:
+ * When cluster is upgraded, the cluster will have both v2 and v3 tables. Now, all the new writes only goes to v3 table.
+ * However, in cases where the row same column is already present in v2 table, for gauge metrics, we first delete that
+ * column from v2 table and then write to v3 table. This approach is used to differentiate gauge and count metrics.
+ *
+ * Reading:
+ * When we scan, we scan from both v2 and v3 tables using {@link CombinedMetricsScanner}. For higher resolution
+ * tables, if the same row is divided between v2 and v3 tables, we merge the columns and add the values of
+ * overlapping columns.
```

Comment:
Add: this also works for gauges, because when writing to the v3 table, we delete the corresponding entry in v2 table. Therefore we will never encounter a value in both tables for a gauge.